File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/exception/InvalidBookingRequestException.java
package com.dominikgaller.alpinebooking.booking.core.domain.exception;

/**
 * Thrown when a booking request violates structural or temporal constraints.
 *
 * <p>Maps to HTTP 400 Bad Request at the REST boundary.
 *
 * <p>SDD: See {@code documentation/domain/aggregate-tour-booking.spec.md}, section 2.
 */
public class InvalidBookingRequestException extends RuntimeException {

    public InvalidBookingRequestException(final String message) {
        super(message);
    }
}

File: src/main/kotlin/com/example/service/booking/core/domain/tourbooking/exception/InvalidBookingRequestException.kt
package com.example.service.booking.core.domain.tourbooking.exception

/**
 * Thrown when a booking request violates a creation-time invariant.
 *
 * Maps to HTTP 400 (`coding-style.definition.md` § 6.2).
 */
class InvalidBookingRequestException(
    message: String,
) : RuntimeException(message)
